src/calendar/tests/CalendarDate.spec.js

```diff
@@ -0,0 +1,463 @@
+import React from 'react/addons';
+import moment from 'moment';
+import _ from 'underscore';
+
+import CalendarDate from '../CalendarDate';
+
+import CalendarDatePeriod from '../CalendarDatePeriod';
+import CalendarHighlight from '../CalendarHighlight';
+import CalendarSelection from '../CalendarSelection';
+
+const TestUtils = React.addons.TestUtils;
+
+describe('The CalendarDate Component', function () {
+
+  beforeEach(function () {
+
+    const getCalendarDate = (props) => {
+
+      props = _.extend({
+        date: moment(),
+        firstOfMonth: moment(),
+        dateRangesForDate: function () {
+          return {
+            count: function () {
+              return props.count || 1;
+            },
+            getIn: function (data) {
+              if (data[0] === 0) {
```

Comment:
I believe this would be because I initially struggled to understand the logic in the source code itself. I will go back to this as I know the project a bit better now.